pkg/reconciler/cache/replication/replication_reconcile.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package replication
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+	"github.com/kcp-dev/logicalcluster/v2"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+)
+
+func (c *controller) reconcile(ctx context.Context, grKey string) error {
+	keyParts := strings.Split(grKey, "::")
+	if len(keyParts) != 2 {
+		return fmt.Errorf("incorrect key: %v, expected group.resource::key", grKey)
+	}
+	switch keyParts[0] {
+	case apisv1alpha1.SchemeGroupVersion.WithResource("apiexports").String():
+		return c.reconcileAPIExports(ctx, keyParts[1], apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"))
+	default:
+		return fmt.Errorf("unsupported resource %v", keyParts[0])
+	}
+}
+
+// reconcileAPIExports makes sure that the ApiExport under the given key from the local shard is replicated to the cache server.
+// the replication function handles the following cases:
+//  1. creation of the object in the cache server when the cached object is not found in c.localApiExportLister
+//  2. deletion of the object from the cache server when the original/local object was removed OR was not found in c.localApiExportLister
+//  3. modification of the cached object to match the original one when meta.annotations, meta.labels, spec or status are different
+func (c *controller) reconcileAPIExports(ctx context.Context, key string, gvr schema.GroupVersionResource) error {
+	var cacheApiExport *apisv1alpha1.APIExport
+	var localApiExport *apisv1alpha1.APIExport
+	var cluster logicalcluster.Name
+	var namespace string
+	var apiExportName string
+	var err error
```

Comment:
why all these vars here? Just do `:= below`

Reply:
C77 style :D

Reply:
I was near, it was 78: https://en.wikipedia.org/wiki/C_(programming_language)#K&R_C

Reply:
:)

Reply:
addressed in https://github.com/kcp-dev/kcp/pull/2090/commits/245aada5b5c970f77c0c4435548cff421aa371d2